### PR TITLE
 get task from ID

### PR DESCRIPTION
**Pull request by sguming**
1, Add controller for HTTPpost
2, Update taskitemDTO for more data
3.SImulate getData from Database
image

### CODE REVIEW

File: blotztask-api/Controllers/TaskController.cs [hunk 1]
```diff
@@ -18,5 +18,10 @@ public async Task<IActionResult> GetAllTask()
         {
             return Ok(await _taskService.GetTodoItems());
         }
+        [HttpPost("[action]")]
```
Comment (sol-wizard) [inline blotztask-api/Controllers/TaskController.cs:21]: use http get for any get request

File: blotztask-api/Controllers/TaskController.cs [hunk 2]
```diff
@@ -18,5 +18,10 @@ public async Task<IActionResult> GetAllTask()
...
+        public async Task<IActionResult> GetTaskByID(int ID)
+        {
+            return Ok(await _taskService.GetTaskByID(ID-1));
```
Comment (sol-wizard) [inline blotztask-api/Controllers/TaskController.cs:24]: -1 ?

File: blotztask-api/Services/TaskService.cs [hunk 3]
```diff
@@ -36,5 +37,18 @@ public async Task<List<TaskItemDTO>> GetTodoItems()
             throw;
         }
     }
+    public async Task<TaskItemDTO> GetTaskByID(int ID)
+    {
+        
+        var taskItems = new List<TaskItemDTO>
```
Comment (sol-wizard) [inline blotztask-api/Services/TaskService.cs:43]: nice👍

File: blotztask-api/Controllers/TaskController.cs [hunk 4]
```diff
@@ -18,5 +18,10 @@ public async Task<IActionResult> GetAllTask()
         {
             return Ok(await _taskService.GetTodoItems());
         }
+        [HttpGet("[action]")]
```
Comment (sol-wizard) [inline blotztask-api/Controllers/TaskController.cs:21]: please have a look https://stackoverflow.com/questions/30967822/when-do-i-use-path-parameters-vs-query-parameters-in-a-restful-api

**Review by sol-wizard — DISMISSED** [review]

**Comment by sguming** [comment]
Using path parameter
image

File: blotztask-api/Controllers/TaskController.cs [hunk 5]
```diff
@@ -18,5 +18,10 @@ public async Task<IActionResult> GetAllTask()
         {
             return Ok(await _taskService.GetTodoItems());
         }
+        [HttpGet("{Id}")]
```
Comment (sol-wizard) [inline blotztask-api/Controllers/TaskController.cs:21]: query param should be camcelCase

change Id to id

**Review by sol-wizard — DISMISSED** [review]

**Review by sol-wizard — APPROVED** [review]

**Pull request merged**